cmd/nerdctl/container_run_test.go

```diff
@@ -533,3 +533,34 @@ func TestRunRmTime(t *testing.T) {
 		t.Fatalf("expected to have completed in %v, took %v", deadline, took)
 	}
 }
+
+func TestRunAttachStdin(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("run attach test is not yet implemented on Windows")
+	}
+
+	t.Parallel()
+	base := testutil.NewBase(t)
+	containerName := testutil.Identifier(t)
+
+	const testStr = "test-run-stdio"
+	opts := []func(*testutil.Cmd){
+		testutil.WithStdin(strings.NewReader("echo " + testStr + "\nexit\n")),
+	}
+
+	defer base.Cmd("rm", "-f", containerName).AssertOK()
+	base.Cmd("run", "--rm", "-a", "stdin", "-a", "stdout", "--name", containerName, testutil.CommonImage).CmdOption(opts...).AssertOutExactly(testStr + "\n")
```

Comment:
nit: while checking you can if the string contains the text. Not  needed to check \n

Reply:
You are probably missing a cat at the end of base command
```
echo "test" | docker run -i -a stdin -a stdout ubuntu     

Prints nothing
```

```
 echo "test" | docker run -i -a stdin -a stdout ubuntu cat  
test
```

Reply:
I added a ```-i``` flag and the tests seem to work now. `cat` wasn't necessary since the tests don't use piping for the stdin.